app/helpers/application_helper/button/smart_state_scan.rb

```diff
@@ -2,8 +2,12 @@ class ApplicationHelper::Button::SmartStateScan < ApplicationHelper::Button::Bas
   def check_smart_roles
     my_zone = MiqServer.my_server.my_zone
     MiqServer::ServerSmartProxy::SMART_ROLES.each do |role|
-      unless MiqServer.all.any? { |s| s.has_active_role?(role) && (s.my_zone == my_zone) }
-        @error_message = _("There is no server with the %{role_name} role enabled") % {:role_name => role}
+      unless MiqServer.all.any? { |s|
+        s.has_active_role?(role) &&
+        s.my_zone == (@record.respond_to?(:zone) ? @record.zone : my_zone)
+      }
+        @error_message = _("There is no server with the %{role_name} role enabled") %
+                         {:role_name => role}
```

Comment:
These changes seem to be the cause of codeclimate failures.
Since they seem to be just reformatting, you may want to consider removing them.

Reply:
@movitto - ping.

Can you resolve the codeclimate and Travis issues before the break?